### PR TITLE
reordering definitions for readability

### PR DESCRIPTION
**Pull request by dericed**
attempt to resolve
https://github.com/Matroska-Org/ebml-specification/issues/120

no wording changes, only ordering

### CODE REVIEW

**Comment by dericed** [comment]
Any comment @bastik-tor?

File: specification.markdown [hunk 1]
```diff
@@ -245,6 +245,18 @@ A `Binary Element` MUST declare a length in octets from zero to `VINTMAX`.
 
 The contents of a `Binary Element` should not be interpreted by the `EBML Reader`.
 
+# Terminating Elements
+
+`Null Octets`, which are octets with all bits set to zero, MAY follow the value of a `String Element` or `UTF-8 Element` to serve as a terminator. An `EBML Writer` MAY terminate a `String Element` or `UTF-8 Element` with `Null Octets` in order to overwrite a stored value with a new value of lesser length while maintaining the same `Element Data Size` (this can prevent the need to rewrite large portions of an `EBML Document`); otherwise the use of `Null Octets` within a `String Element` or `UTF-8 Element` is NOT RECOMMENDED. An `EBML Reader` MUST consider the value of the `String Element` or `UTF-8 Element` to be terminated upon the first read `Null Octet` and ignore any following data within that `Element`. If an `EBML Writer` terminates the value of a `String Element` or `UTF-8 Element` with a `Null Octet` then any subsequent octets of the `Element` MUST also be `Null Octets`. A string value and a copy of that string value terminated by one or more `Null Octets` are semantically equal.
```
Comment (robUx4) [inline specification.markdown]: You may put these changes in a separate patch. It's not just reordering of elements and so requires a different kind of review that may leave this patch up for a lot longer.

Comment (mbunkus) [inline specification.markdown]: I guess this work was simply based on the wrong branch and that the section about terminating the element wasn't supposed to be in this patch. There is a separate pull request #141 for that, after all.

**Comment by bastik-tor** [comment]
@dericed There is nothing I could add to this. At least I have nothing to object against.

Thank you.

File: specification.markdown (hunk 1 above)
Comment (dericed) [inline specification.markdown]: Yes this was an error. I rebased the PR to remove this redundancy.

**Comment by dericed** [comment]
@robUx4 noticed that some non-ordering changes were included in the patch. This was an error, the PR is now rebased to only focus on re-ordering definitions. It should be now ready to merge.

**Comment by mbunkus** [comment]
LGTM now. As others LGTM'ed it before, I'm going to merge it now.
